src/streamlink/plugins/pluto.py

```diff
@@ -1,99 +1,160 @@
-            data = self.session.http.json(
-                res,
-                schema=validate.Schema(
-                    {'categories': validate.all(
-                        [{'items': self._schema_media(m['slug_movies'])}],
-                        validate.filter(lambda k: k['items'] is not None))},
-                    validate.get('categories'),
-                    validate.get(0),
-                    validate.any(None, validate.get('items')),
-                ),
-            )
+        if m["slug_live"]:
+            data = self._get_api_data("channelSlug", m["slug_live"])
 
-        log.trace(f'{data!r}')
-        if data is None or not data.get('stitched'):
-            return
+            media = data.get("EPG")
+            if media and media[0]["slug"] == m["slug_live"]:
+                self.id = media[0]["id"]
+                self.title = media[0]["name"]
+                path = media[0]["stitched"]["path"]
+        elif m["slug_series"] and m["slug_episode"]:
+            data = self._get_api_data("episodeSlugs", m["slug_series"], filter=m["slug_episode"])
+
+            media = data.get("VOD")
+            if media and media[0]["slug"] == m["slug_series"]:
+                self.author = media[0]["name"]
+                self.category = media[0]["genre"]
+                if "seasons" in media[0] and media[0]["seasons"][0]["episodes"][0]["slug"] == m["slug_episode"]:
+                    self.id = media[0]["seasons"][0]["episodes"][0]["_id"]
+                    self.title = media[0]["seasons"][0]["episodes"][0]["name"]
+                    path = media[0]["seasons"][0]["episodes"][0]["stitched"]["path"]
 
-        self.title = data['name']
-        stream_url_no_sid = data['stitched']['urls'][0]['url']
-        device_id = str(uuid4())
-        stream_url = update_qsd(stream_url_no_sid, {
-            'deviceId': device_id,
-            'sid': device_id,
-            'deviceType': 'web',
-            'deviceMake': 'Firefox',
-            'deviceModel': 'Firefox',
-            'appName': 'web',
-        })
-
-        self.session.set_option('ffmpeg-fout', 'mpegts')
-        for q, s in HLSStream.parse_variant_playlist(self.session, stream_url).items():
-            yield q, MuxedStream(self.session, s)
+        elif m["slug_movies"]:
+            data = self._get_api_data("episodeSlugs", m["slug_movies"])
+
+            media = data.get("VOD")
+            if media and media[0]["slug"] == m["slug_movies"]:
+                self.category = media[0]["genre"]
+                self.id = media[0]["id"]
+                self.title = media[0]["name"]
+                path = media[0]["stitched"]["path"]
+
+        if data and path:
+            log.trace(f"data={data!r}")
+            log.debug(f"path={path}")
+
+            return self._get_playlist(
+                data["servers"]["stitcher"],
+                path,
+                data["stitcherParams"],
+                data["sessionToken"],
+            )
```

Comment:
This should get simplified. All these lookups are redundant and barely legible.

Since you require `data` and `path` to not be `None` and since `path` only gets set in these nested if-statements, just invert the conditions and return `None`, so that all this nesting can be avoided. Also, why is the `seasons` property optional in the schema but mandatory here for returning streams (otherwise `path` is `None`)?

Example (untested):
```py
    @staticmethod
    def _get_media_data(data, key, slug):
        media = data.get(key)
        if media and media[0]["slug"] == slug:
            return media[0]

    def _get_streams(self):
        m = self.match.groupdict()
        if m["slug_live"]:
            data = self._get_api_data("channelSlug", m["slug_live"])
            media = self._get_media_data(data, "EPG", m["slug_live"])
            if not media:
                return

            self.id = media["id"]
            self.title = media["name"]
            path = media["stitched"]["path"]

        elif m["slug_series"] and m["slug_episode"]:
            data = self._get_api_data("episodeSlugs", m["slug_series"], filter=m["slug_episode"])
            media = self._get_media_data(data, "VOD", m["slug_series"])
            if not media or "seasons" not in media:
                return
            episode = media["seasons"][0]["episodes"][0]
            if episode["slug"] != m["slug_episode"]:
                return

            self.author = media["name"]
            self.category = media["genre"]
            self.id = episode["_id"]
            self.title = episode["name"]
            path = episode["stitched"]["path"]

        elif m["slug_movies"]:
            data = self._get_api_data("episodeSlugs", m["slug_movies"])
            media = self._get_media_data(data, "VOD", m["slug_movies"])
            if not media:
                return

            self.category = media["genre"]
            self.id = media["id"]
            self.title = media["name"]
            path = media["stitched"]["path"]

        else:
            return

        log.trace(f"data={data!r}")
        log.debug(f"path={path}")

        return self._get_playlist(
            data["servers"]["stitcher"],
            path,
            data["stitcherParams"],
            data["sessionToken"],
        )
```

Reply:
That does look a lot cleaner, thanks.

> Also, why is the seasons property optional in the schema but mandatory here for returning streams (otherwise path is None)?

The `seasons` property does not appear in the returned API data for movies, so it's optional there, but when looking up series based VoD content it's mandatory to at least get the correct `path` property.  Does that seem reasonable?